src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.DotNet.MSIdentity.AuthenticationParameters;
+using Microsoft.DotNet.MSIdentity.Properties;
+using Microsoft.DotNet.MSIdentity.Tool;
+
+namespace Microsoft.DotNet.MSIdentity.CodeReaderWriter
+{
+    internal class ProjectModifier
+    {
+        private readonly ProvisioningToolOptions _toolOptions;
+        private readonly ApplicationParameters _appParameters;
+        private readonly IConsoleLogger _consoleLogger;
+
+        public ProjectModifier(ApplicationParameters applicationParameters, ProvisioningToolOptions toolOptions, IConsoleLogger consoleLogger)
+        {
+            _toolOptions = toolOptions ?? throw new ArgumentNullException(nameof(toolOptions));
+            _appParameters = applicationParameters ?? throw new ArgumentNullException(nameof(applicationParameters));
+            _consoleLogger = consoleLogger ?? throw new ArgumentNullException(nameof(consoleLogger));
+        }
+
+        /// <summary>
+        /// Added "Microsoft identity platform" auth to base or empty C# .NET Core 3.1, .NET 5 and above projects.
+        /// Includes adding PackageReferences, modifying Startup.cs, and Layout.cshtml(not yet) changes.
+        /// </summary>
+        /// <param name="projectType"></param>
+        /// <returns></returns>
+        public async Task AddAuthCodeAsync()
+        {
+            CodeModifierConfig? codeModifierConfig = GetCodeModifierConfig();
+            //Get CodeModifierConfig from CodeModifierConfigs folder.
+            if (codeModifierConfig != null &&
+                codeModifierConfig.Files != null &&
+                !string.IsNullOrEmpty(_toolOptions.ProjectFilePath))
```

Comment:
>!string.IsNullOrEmpty(_toolOptions.ProjectFilePath) [](http://example.com/codeflow?start=16&length=51)

check this first and exit before doing GetCodeModifierConfig

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593